nidirect_contacts/src/Plugin/Field/FieldFormatter/GMapsLazyLoadFormatter.php

```diff
@@ -0,0 +1,248 @@
+      '#default_value' => $this->getSetting('zoom'),
+    ];
+
+    // Options to render various placeholders until the container is visible
+    // and the JS has loaded the map.
+    $form['placeholder'] = [
+      '#title' => $this->t('Placeholder'),
+      '#type' => 'select',
+      '#options' => [
+        'empty' => t('Empty'),
+        'link' => t('Link to Google map'),
+        'static_map' => t('Static map'),
+      ],
+      '#default_value' => $this->getSetting('placeholder'),
+    ];
+
+    // Specify the text for the link placeholder.
+    $form['link_text'] = [
+      '#title' => $this->t('Link text'),
+      '#type' => 'textfield',
+      '#default_value' => $this->getSetting('link_text'),
+      '#states' => [
+        'visible' => [
+          ':input[name="fields[field_location][settings_edit_form][settings][placeholder]"]' => ['value' => 'link'],
+        ],
+      ],
+    ];
+
+    // Static map width in pixels.
+    $form['map_width'] = [
+      '#title' => $this->t('Map width (px)'),
+      '#type' => 'number',
+      '#min' => 60,
+      '#max' => 2000,
+      '#default_value' => $this->getSetting('map_width'),
+      '#states' => [
+        'visible' => [
+          ':input[name="fields[field_location][settings_edit_form][settings][placeholder]"]' => ['value' => 'static_map'],
+        ],
+      ],
+    ];
+
+    // Static map height in pixels.
+    $form['map_height'] = [
+      '#title' => $this->t('Map height (px)'),
+      '#type' => 'number',
+      '#min' => 60,
+      '#max' => 2000,
+      '#default_value' => $this->getSetting('map_height'),
+      '#states' => [
+        'visible' => [
+          ':input[name="fields[field_location][settings_edit_form][settings][placeholder]"]' => ['value' => 'static_map'],
+        ],
+      ],
+    ];
+
+    return $form + parent::settingsForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsSummary() {
+    $summary = [];
+    $summary[] = $this->t(
+      'Map type: @maptype <br> Zoom: @zoom <br> Placeholder: @placeholder', [
+        '@maptype' => $this->getSetting('map_type'),
+        '@zoom' => $this->getSetting('zoom'),
+        '@placeholder' => $this->getSetting('placeholder'),
+      ]
+    );
+
+    // Additional summary information for the static map.
+    if ($this->getSetting('placeholder') === 'static_map') {
+      $summary[] .= $this->t(
+        'Width: @widthpx Height: @heightpx', [
+          '@width' => $this->getSetting('map_width'),
+          '@height' => $this->getSetting('map_height'),
+        ]
+      );
+    }
+
+    return $summary;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $elements = [];
+
+    $formatter_settings = $this->getSettings();
+    $google_provider = \Drupal::service('plugin.manager.geolocation.mapprovider')->getMapProvider('google_maps');
+    $google_settings = \Drupal::config('geolocation_google_maps.settings');
```

Comment:
Could Dependency Injection be used here ?

Reply:
True, my bad